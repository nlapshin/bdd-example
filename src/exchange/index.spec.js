const Exchange = require('.')

describe('exchange.getData', () => {
  test('it should return list of exchange with Diff field', async () => {
    const exchange = new Exchange()
    jest.spyOn(exchange, 'fetchData').mockImplementation(() => {

    return Promise.resolve([
      {
        "ID": "R01010",
        "NumCode": "036",
        "CharCode": "AUD",
        "Nominal": 1,
        "Name": "Австралийский доллар",
        "Value": 53.9159,
        "Previous": 55.8293
      },
      {
          "ID": "R01020A",
          "NumCode": "944",
          "CharCode": "AZN",
          "Nominal": 1,
          "Name": "Азербайджанский манат",
          "Value": 43.2382,
          "Previous": 42.1171
      }
    ])});

    const result = await exchange.getData()

    expect(result).toMatchSnapshot();
  })
})
