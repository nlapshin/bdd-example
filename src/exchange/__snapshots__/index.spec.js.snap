// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`exchange.getData it should return list of exchange with Diff field 1`] = `
Array [
  Object {
    "CharCode": "AUD",
    "Diff": -1.9134,
    "ID": "R01010",
    "Name": "Австралийский доллар",
    "Nominal": 1,
    "NumCode": "036",
    "Previous": 55.8293,
    "Value": 53.9159,
  },
  Object {
    "CharCode": "AZN",
    "Diff": 1.1211,
    "ID": "R01020A",
    "Name": "Азербайджанский манат",
    "Nominal": 1,
    "NumCode": "944",
    "Previous": 42.1171,
    "Value": 43.2382,
  },
]
`;
